lib/livebook_web/live/hub/edit_live.ex

```diff
@@ -115,6 +115,27 @@ defmodule LivebookWeb.Hub.EditLive do
      |> put_flash(:success, "Secret #{name} deleted successfully")}
   end
 
+  def handle_info({:file_system_created, _}, socket) do
+    {:noreply,
+     socket
+     |> push_navigate(to: ~p"/hub/#{socket.assigns.hub.id}")
+     |> put_flash(:success, "File storage added successfully")}
+  end
+
+  def handle_info({:file_system_updated, _}, socket) do
+    {:noreply,
+     socket
+     |> push_navigate(to: ~p"/hub/#{socket.assigns.hub.id}")
+     |> put_flash(:success, "File storage updated successfully")}
+  end
+
+  def handle_info({:file_system_deleted, _}, socket) do
+    {:noreply,
+     socket
+     |> push_navigate(to: ~p"/hub/#{socket.assigns.hub.id}")
+     |> put_flash(:success, "File storage deleted successfully")}
```

Comment:
The way this is implemented, if someone else is on the page and we create a file storage/secret, then the notification will appear to everyone and the page will be reloaded. Perhaps we should make these broadcasts look like this:

```
{:file_system_deleted, self(), metadata}
```

And we only match on those events, in this particular cases, if the pid in the message matches `self()`?

Reply:
Btw, for the session messages, we check if the hub_id matches, but that's not enough to avoid the issue above. If we always check the pid, then we don't need to check the hub_id. :)

Reply:
Perhaps we should make this change in another PR though.

Reply:
Ah, there is a simpler way to solve this. Instead of listening to the pubsub events, have the session and file storage components do the redirect and set the flash messages themselves. Would that work? If that doesn't work, we can have the components send a message like this: `send(self(), {:redirect, flash, url})`, which we match here, instead of pubsub.

Reply:
Btw, the change above means that the page no longer automatically updates if someone else adds a secret/filesystem, but that's fine by me. If we want to keep the automatic updates, we also can. In this case, we should change all of these `handle_info` callbacks to something like this:

    def handle_info({_event, _message}, socket) do
      hub = Hubs.fetch_hub!(socket.assigns.hub.id)
      {:noreply, assign(socket, :hub, hub)}
    end

This way there is no flash message and no page redirect. These would be done by the session/form components themselves.

Reply:
It didn't refresh the second session on the hubs page. I'll try the `send/2` approach.